DS-3290: Add unit tests for 100,000 case regression

mwmclean opened the pull request.

jrwishart commented on tests/testthat/test-regression100Kcases.R:
```diff
@@ -0,0 +1,125 @@
+
+test_that("Linear regression",
+{
+    expect_is(Regression(log.price~.-region_weight, data = vehicles),
+              "LinearRegression")
+    ## Effects Plot doesn't like "." in formula
+    expect_error(print(Regression(log.price ~ drive+condition+log.odometer+region+year,
+                   data = vehicles,
+                   weights = region_weight,
+                   output = "Effects Plot")), NA)
+    expect_error(print(Regression(log.price~.-region_weight,
+                   data = vehicles,
+                   weights = region_weight,
+                   output = "ANOVA")), NA)
+
+    expect_is(Regression(log.price~.-region_weight, data = vehicles,
+                         weights = region_weight),
+              "LinearRegression")
```
Could perhaps make these tests faster by creating two regression objects and then referencing them in print statements instead of running the regression 4 times. The expanded formula could be used since the `.` is tested in `flipRegression`

jrwishart replied:
I also got this when running the file locally. Could perhaps update the tests to allow these warnings to appear.
![image](https://user-images.githubusercontent.com/4605715/117736584-4f689280-b23b-11eb-85f5-43123f1b978a.png)

mwmclean replied:
I have restructured this test_that block and added expectations for the warnings.

jrwishart commented on tests/testthat/test-regression100Kcases.R:
```diff
@@ -0,0 +1,125 @@
+context("Regression with 100,000 cases")
+
+vehicles <- readRDS("vehicles100kNoNA.rds")
```
~This data file has 5 predictors. The spec wanted 10 and 20 predictors. Could augment the `data.frame` with 15 more predictors via simulation?~

Ignore this, can see the simulation code below

jrwishart commented on tests/testthat/test-regression100Kcases.R:
```diff
@@ -0,0 +1,125 @@
+
+test_that("Multinomial response",
+{
+    expect_is(Regression(year~.-region_weight,
+                         data = vehicles, type = "Multinomial Logit"),
+              "MultinomialLogitRegression")
+
```
weighted multinomial logit is missing in the tests.

mwmclean replied:
Added

jrwishart reviewed (commented):
Suggest adding more tests for crosstab interaction and covering the missing data options. Stacking data would also likely blow up the memory some more if there is a requirement for stacking. That isn't explicitly in the spec though.

mwmclean commented:
> Suggest adding more tests for crosstab interaction and covering the missing data options. Stacking data would also likely blow up the memory some more if there is a requirement for stacking. That isn't explicitly in the spec though.

I added tests for interaction, dummy variable adjustment, and multiple imputation. I added `skip_on_travis()` for the multiple imputation test as I think it will eat up too much of our travis time limit and there is already an long blog test.
